File: Leetcode 0334 Increasing Triplet Subsequence.go
// 2025/01/14

func increasingTriplet(nums []int) bool {
    min1 := math.MaxInt64 // 初始化為 Go 的最大整數
    min2 := math.MaxInt64

    for _, n := range nums{
        if (n <= min1){
            min1 = n
        }else if (n<=min2){
            min2 = n
        }else{
            return true
        }
    }
    return false
}

File: Leetcode 0334 Increasing Triplet Subsequence.py
# 2025/01/14

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return False
        ans = [float('inf'), float('inf')]
        for i in range(0, len(nums)):
            if ans[0] >= nums[i]:
                ans[0] = nums[i]
            elif ans[1] >= nums[i]:
                ans[1] = nums[i]
            else:
                return True
        return False
    
